nix/jjui.nix

```diff
@@ -0,0 +1,78 @@
+{ lib, ... }:
+{
+  perSystem =
+    { pkgs, ... }:
+    let
+      buildInputs = [ pkgs.jujutsu ]; # Runtime
+      nativeBuildInputs = [ pkgs.makeWrapper ]; # Buildtime
+
+      jjui = pkgs.buildGoModule {
+        inherit buildInputs nativeBuildInputs;
+
+        pname = "jjui";
+        version = "dev";
+
+        src = lib.fileset.toSource {
+          root = ./..;
+          fileset = lib.fileset.unions [
+            ./../go.mod
+            ./../go.sum
+            ./../cmd
+            ./../internal
+            ./../test
+          ];
+        };
+        vendorHash = lib.strings.trim (builtins.readFile ./vendor-hash);
+        doCheck = false;
+
+        postInstall = ''
+          wrapProgram $out/bin/jjui \
+            --prefix PATH : ${pkgs.lib.makeBinPath [ pkgs.jujutsu ]}
+        '';
+
+        meta = {
+          description = "A Text User Interface (TUI) designed for interacting with the Jujutsu version control system";
+          homepage = "https://github.com/idursun/jjui";
+          license = pkgs.lib.licenses.mit;
+          maintainers = [ "idursun" ];
+          platforms = pkgs.lib.platforms.unix;
+          mainProgram = "jjui";
+        };
+      };
+
+      update-vendor-hash = pkgs.writeShellApplication {
```

Comment:
lets move this helper app into its own file (another flake-parts module) since it is not related to packaging jjui itself.